Add Tyk Gateway and Tyk Dashboard Release Notes 5.2.2 (#3384)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-dashboard/release-notes/version-5.2.md

---

## 5.2.2 Release Notes 

##### Release Date 31 Oct 2023

#### Breaking Changes
This release has no breaking changes.

#### Deprecations
There are no deprecations in this release.

#### Upgrade instructions
If you are using a 5.2.x version, we advise you to upgrade ASAP to this latest release. If you are on an older version, you should skip 5.2.0 and upgrade directly to this release.

#### Release Highlights
This release primarily focuses on bug fixes. 
For a comprehensive list of changes, please refer to the detailed [changelog]({{< ref "#Changelog-v5.2.2">}}) below.

#### Downloads

> - [docker image TO ADD](https://hub.docker.com/layers/tykio/tyk-dashboard/v5.2.1/images/sha256-2f9d8af0e57f7fe4afb618dcf34772c001104dc0ec62a27541d12dc9ae90d5c8?context=explore)

Author: - [docker image to pull](https://hub.docker.com/layers/tykio/tyk-dashboard/v5.2.2/images/sha256-c6e701e270ebb2fed815483723375c454d0479ae41b5be2e1a6198b8d1e1a154?context=explore)

---

## 5.2.2 Release Notes 

> ##### Release Date 27 Oct 2023

Author: ```suggestion
##### Release Date 31 Oct 2023
```

#### Breaking Changes
This release has no breaking changes.

#### Deprecations
There are no deprecations in this release.

#### Upgrade instructions
If you are using a 5.2.x version, we advise you to upgrade ASAP to this latest release. If you are on an older version, you should skip 5.2.0 and upgrade directly to this release.

#### Release Highlights
This release primarily focuses on bug fixes. 

> For a comprehensive list of changes, please refer to the detailed [changelog]({{< ref "#Changelog-v5.2.0">}}) below.

Reviewer: Should this link to #Changelog-v5.2.2 instead?

```suggestion
For a comprehensive list of changes, please refer to the detailed [changelog]({{< ref "#Changelog-v5.2.2">}}) below.
```

---

## 5.2.2 Release Notes 

##### Release Date <<TO UPDATE>> 2023

#### Breaking Changes
This release has no breaking changes.

#### Deprecation
There are no deprecations in this release.

#### Upgrade instructions
If you are on a 5.2.0 we advise you to upgrade ASAP and if you are on an older version skip 5.2.0 and upgrade directly to this release.

#### Release Highlights
This release primarily focuses on bug fixes. 
For a comprehensive list of changes, please refer to the detailed [changelog]({{< ref "#Changelog-v5.2.0">}}) below.

#### Downloads
- [docker image TO ADD](TO ADD)

#### Changelog {#Changelog-v5.2.2}

#### Added

> - Added a new Dashboard configuration option `Allow_Unsafe_Oas`. This permits the modification of Tyk OAS APIs via the Tyk Classic API endpoints. This is not a recommended action due to the risk of inconsistent behaviour and potential for breaking changes while Tyk OAS is in [Early Access]({{< ref "frequently-asked-questions/using-early-access-features" >}}). However, it is provided for early adopters and will be deprecated later. Please note that `Allow_Unsafe_Oas` must be set to `True` to use Tyk OAS APIS with Tyk Sync.

Reviewer 2: @andyo-tyk Should this be removed 'Please note that `Allow_Unsafe_Oas` must be set to `True` to use Tyk OAS APIS with Tyk Sync.'?

Reviewer: ```suggestion
- Added a new Dashboard configuration option `allow_unsafe_oas`. This permits the modification of Tyk OAS APIs via the Tyk Classic API endpoints. This is not a recommended action due to the risk of inconsistent behaviour and potential for breaking changes while Tyk OAS is in [Early Access]({{< ref "frequently-asked-questions/using-early-access-features" >}}). However, it is provided for early adopters and will be deprecated later.
```

#### Fixed

- Fixed an issue with MongoDB where Tyk could incorrectly grant access to an API using a key after that API had been deleted from the associated policy. This was due to the policy cleaning operation in MongoDB installations, which Tyk interpreted as granting access to the original API. With this fix, the policy cleaning operation will not remove the final (Deleted) API from the policy; Tyk recognises that the API record is invalid and denies granting access rights to the key.

> - Fixed an issue in The Tyk Dashboard where a user did not correctly inherit all permissions from their user group, permitting visibility of identity management.

Reviewer: ```suggestion
- Fixed an issue in The Tyk Dashboard where a user did not correctly inherit all permissions from their user group, incorrectly permitting visibility of Identity Management.
```

> - Fixed an issue where Tyk would not store the *Policy Id* in the *Api Definition* for a policy that did not exist. When using *JWT Authentication*, the *JWT Default Policy Id* is stored in the *Api Definition*. If this policy had not been created in Tyk at the time the *Api Definition* was created, Tyk Dashboard would invalidate the field in the *Api Definition*. When the policy was later created, there would be no reference to it from the *Api Definition*. This was a particular issue when using *Tyk Operator* to manage the creation of assets on Tyk.

Reviewer: API should always be in upper case - it is an abbreviation.

```suggestion
- Fixed an issue where Tyk would not store the *Policy Id* in the *API Definition* for a policy that did not exist. When using *JWT Authentication*, the *JWT Default Policy Id* is stored in the *API Definition*. If this policy had not been created in Tyk at the time the *API Definition* was created, Tyk Dashboard would invalidate the field in the *API Definition*. When the policy was later created, there would be no reference to it from the *API Definition*. This was a particular issue when using *Tyk Operator* to manage the creation of assets on Tyk.
```

- Fixed an issue in the Dashboard *Service Uptime* page where the number of success hits was being incorrectly reported as the total number of hits, inclusive of failures. After this fix, the *Success Column* displays only the number of success hits.

- Fixed the following high priority Cves identified in the dashboard to provide increased protection against security vulnerabilities:

>     - [Cve-2022-33082](https://nvd.nist.gov/vuln/detail/CVE-2022-33082)

Reviewer: All of these should be capitalised, I don't understand why they've been made mixed case?
```suggestion
    - [CVE-2022-33082](https://nvd.nist.gov/vuln/detail/CVE-2022-33082)
```

    - [Cve-2022-28946](https://nvd.nist.gov/vuln/detail/CVE-2022-28946)
    - [Cve-2021-23409](https://nvd.nist.gov/vuln/detail/CVE-2021-23409)
    - [Cve-2021-23351](https://nvd.nist.gov/vuln/detail/CVE-2021-23351)
    - [Cve-2023-28119](https://nvd.nist.gov/vuln/detail/CVE-2023-28119)
    - [Cve-2022-21698](https://nvd.nist.gov/vuln/detail/CVE-2022-21698)
    - [Cve-2020-26160](https://nvd.nist.gov/vuln/detail/CVE-2020-26160)
    - [Cve-2019-19794](https://nvd.nist.gov/vuln/detail/CVE-2019-19794)
    - [Cve-2010-0928](https://nvd.nist.gov/vuln/detail/CVE-2010-0928)
    - [Cve-2007-6755](https://nvd.nist.gov/vuln/detail/CVE-2007-6755)
    - [Cve-2018-5709](https://nvd.nist.gov/vuln/detail/CVE-2018-5709)

> - Fixed an issue encountered with *Azure Saml2.0 Identity Provider* that was preventing users from authenticating.

Reviewer: Again, SAML has been made mixed case :(

```suggestion
- Fixed an issue encountered with *Azure SAML2.0 Identity Provider* that was preventing users from authenticating.
```

- Fixed an issue encountered with the *API Designer* where fields defined in *Uptime_Tests.Check_List* were not correctly handled. Uptime tests can now be safely handled for *Tyk Classic APIs* in the *Raw API Definition* editor.

> - Fixed a security vulnerability with the Tyk Dashboard API where the `Url_Version` query parameter was a potential target for a SQL injection attack. This addresses [Cve-2023-42284](https://nvd.nist.gov/vuln/detail/CVE-2023-2284).

Reviewer: `url_version` is a parameter name - it's lower case - not sure why it's been made mixed case here?

```suggestion
- Fixed a security vulnerability with the Tyk Dashboard API where the `url_version` query parameter was a potential target for a SQL injection attack. This addresses [CVE-2023-42284](https://nvd.nist.gov/vuln/detail/CVE-2023-2284).
```

> - Fixed a security vulnerability with the Tyk Dashboard API where the `Url_Version` query parameter was a potential target for a SQL injection attack. This addresses [Cve-2023-42283](https://nvd.nist.gov/vuln/detail/CVE-2023-40283).

Reviewer: ```suggestion
- Fixed a security vulnerability with the Tyk Dashboard API where the `url_version` query parameter was a potential target for a SQL injection attack. This addresses [CVE-2023-42283](https://nvd.nist.gov/vuln/detail/CVE-2023-40283).
```

#### Updated

> - Renamed The Licence Limit To Licence Entitlement. Hide The Licence Entitlement Line From The Chart In Case Its Value Is Set To 0.

Reviewer: Weirdly this comment has every word capitalised!!

```suggestion
- On Tyk Dashboard's Licensing Statistics screen we have renamed the Licence Limit to Licence Entitlement. We've also improved the experience when there is no limit in the licence by hiding the Licence Entitlement line if no limit is set.
```
